### PR TITLE
Remove properties from tsconfig when converting to jsconfig

### PR DESCRIPTION
**Pull request by jtoar**
Closes https://github.com/redwoodjs/redwood/issues/2020. Thoughts on if this is the right approach and/or if there's any other properties we'd like to remove from the `tsconfig.json`s while we're at it?

Also it turns out the `tsconfig.json`s in `create-redwood-app/template` weren't valid JSON (trailing commas). VS Code doesn't seem to care though; prettier may be interfering with it.

### CODE REVIEW

File: packages/internal/src/ts2js.ts [hunk 1]
```diff
@@ -48,25 +50,33 @@ export const convertTsFilesToJs = (cwd: string, files: string[]) => {
...
-    fs.renameSync(
-      path.join(cwd, 'scripts/tsconfig.json'),
-      path.join(cwd, 'scripts/jsconfig.json')
-    )
-  }
+/**
+ * @param {string} cwd
+ */
+export function convertTsConfigsToJsConfigs(cwd = getPaths().base) {
+  const tsConfigs = [
```
Comment (dac09) [inline packages/internal/src/ts2js.ts:59]: Wouldn't it be better to use a glob to search for all tsconfigs in the generated project?

File: packages/internal/src/ts2js.ts [hunk 2]
```diff
@@ -48,25 +50,33 @@ export const convertTsFilesToJs = (cwd: string, files: string[]) => {
...
+    'api/tsconfig.json',
+    'web/tsconfig.json',
+    'scripts/tsconfig.json',
+  ]
+
+  tsConfigs
+    .filter((tsConfig) => fs.existsSync(path.join(cwd, tsConfig)))
```
Comment (dac09) [inline packages/internal/src/ts2js.ts:66]: If we use a glob, we can remove this extra check

File: packages/internal/src/ts2js.ts [hunk 3]
```diff
@@ -48,25 +50,33 @@ export const convertTsFilesToJs = (cwd: string, files: string[]) => {
...
+    .map((tsConfig) => readJSONSync(path.join(cwd, tsConfig)))
+    .map(convertTsConfigToJsConfig)
+    .forEach((jsConfig, i) => {
+      writeJSONSync(
```
Comment (dac09) [inline packages/internal/src/ts2js.ts:70]: Not a big deal but the jsconfig json produced isn't formatted nicely. I wonder if there is the equivalent of `JSON.stringify({ a:1, b:2, c:3 }, null, 4)` to format it a little better with writeJsonSync?

File: packages/internal/src/ts2js.ts (hunk 1 above)
Comment (jtoar) [reply packages/internal/src/ts2js.ts]: Done in https://github.com/redwoodjs/redwood/pull/4263/commits/941557b1e1eebb449c475eed04ce331c361481bb.

File: packages/internal/src/ts2js.ts (hunk 2 above)
Comment (jtoar) [reply packages/internal/src/ts2js.ts]: Done in https://github.com/redwoodjs/redwood/pull/4263/commits/941557b1e1eebb449c475eed04ce331c361481bb.

File: packages/internal/src/ts2js.ts [hunk 4]
```diff
@@ -48,25 +50,30 @@ export const convertTsFilesToJs = (cwd: string, files: string[]) => {
...
-      path.join(cwd, 'scripts/tsconfig.json'),
-      path.join(cwd, 'scripts/jsconfig.json')
-    )
-  }
+/**
+ * @param {string} cwd
+ */
+export function convertTsConfigsToJsConfigs(cwd = getPaths().base) {
+  // Should find 'api/tsconfig.json`, `web/tsconfig.json`, and `scripts/tsconfig.json`.
+  const tsConfigs = fg.sync('*/tsconfig.json', { cwd, absolute: true })
```
Comment (dac09) [inline packages/internal/src/ts2js.ts:60]:
```suggestion
  const tsConfigs = fg.sync('*/tsconfig.json', { cwd, absolute: true, ignore: '**/node_modules'
 })
```

File: packages/internal/src/ts2js.ts (hunk 3 above)
Comment (jtoar) [reply packages/internal/src/ts2js.ts:66]: Ah missed that—it looks like the [`spaces` option](https://github.com/jprichardson/node-fs-extra/blob/master/docs/writeJson-sync.md#writejsonsyncfile-object-options) does what we want.

Done in https://github.com/redwoodjs/redwood/pull/4263/commits/9a19d8f4556efe87c5a797072252739de234d6b6 and https://github.com/redwoodjs/redwood/pull/4263/commits/3b23d713daca452d79e6b7f2a832d08f27dc62e9 (left a console log).

File: packages/internal/src/ts2js.ts [hunk 5]
```diff
@@ -57,8 +57,11 @@ export const convertTsFilesToJs = (cwd: string, files: string[]) => {
  */
 export function convertTsConfigsToJsConfigs(cwd = getPaths().base) {
   // Should find 'api/tsconfig.json`, `web/tsconfig.json`, and `scripts/tsconfig.json`.
-  const tsConfigs = fg.sync('*/tsconfig.json', { cwd, absolute: true, ignore: '**/node_modules'
- })
+  const tsConfigs = fg.sync('*/tsconfig.json', {
+    cwd,
+    absolute: true,
+    ignore: ['node_modules'],
```
Comment (dac09) [inline packages/internal/src/ts2js.ts:63]: I'm assuming you've checked this, but maybe it should be `**/node_modules`?

Comment (jtoar) [reply packages/internal/src/ts2js.ts:63]: Yeah when I run it only I find the three tsconfigs we expect. I'm not using the globstar, only the single start (not sure what it's called—`*`), so I'm only matching one directory, one-level deep. So even if we didn't ignore `node_modules`, we wouldn't be searching through every package inside `node_modules`. I.e. to get a match in node_modules if we didn't ignore it, there'd have to be a package called `tsconfig.json`.

**Review by dac09 — APPROVED** [review]

**Comment by jtoar** [comment]
I can reproduce CI hanging locally. Investigating.

**Comment by jtoar** [comment]
Superseded by https://github.com/redwoodjs/redwood/pull/8191

**Pull request closed without merging**